core/services/relay/evm/contract_binding.go

```diff
@@ -0,0 +1,93 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/google/uuid"
+
+	commontypes "github.com/smartcontractkit/chainlink-common/pkg/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+)
+
+type filterRegisterer struct {
+	pollingFilter logpoller.Filter
+	filterLock    sync.Mutex
+	// registerCalled is used to determine if Register was called during Chain Reader service Start.
+	// This is done to avoid calling Register while the service is not running because log poller is most likely also not running.
+	registerCalled bool
+}
+
+// contractBinding stores read bindings and manages the common contract event filter.
+type contractBinding struct {
+	name string
+	// filterRegisterer is used to manage polling filter registration for the common contract filter.
+	// The common contract filter should be used by events that share filtering args.
+	filterRegisterer
```

Comment:
contract binding can also have multiple common filters defined, but this creates more complexity and I see no need for this right now